internal/langserver/handlers/did_change_watched_files.go

```diff
@@ -0,0 +1,49 @@
+package handlers
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+
+	ilsp "github.com/hashicorp/terraform-ls/internal/lsp"
+	"github.com/hashicorp/terraform-ls/internal/protocol"
+	lsp "github.com/hashicorp/terraform-ls/internal/protocol"
+	"github.com/hashicorp/terraform-ls/internal/terraform/ast"
+)
+
+func (svc *service) DidChangeWatchedFiles(ctx context.Context, params lsp.DidChangeWatchedFilesParams) error {
+	for _, change := range params.Changes {
+		URI := string(change.URI)
+		dir, err := os.Stat(URI)
+		if err != nil {
+			return fmt.Errorf("Unable to update %s. Failed to open directory, error: %w", URI, err)
+		}
+
+		filename := path.Base(URI)
+		if !dir.IsDir() && !ast.IsModuleFilename(filename) {
+			// TODO: raise warning; unsupported file
+			return fmt.Errorf("Unable to update %s, filetype not supported.", filename)
```

Comment:
Since we want to show a warning/error notification, do you think we should return a custom error type and also emit a `window/showMessage` here?